### PR TITLE
Add Rewards self-custody provider UI components

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/34949

- Added a boolean pref to track whether the user has dismissed the self-custody invitation.
- Added the following methods to `chrome.braveRewards` in order to support the self-custody inviation on the NTP and the Rewards panel.
  - `onSelfCustodyInviteDismissed`
  - `selfCustodyInviteDismissed`
  - `dismissSelfCustodyInvite`
- Updated front-end code on the NTP and the Rewards panel to support the self-custody invitation.
- Updated the Rewards page to hide the AC and Tips sections for self-custody providers.
- Updated the connect wallet modal to support self-custody providers.
- Updated the connect wallet modal to show a loading spinner and an error message during the initial phase of linking (prior to redirecting the user).
- Updated the tipping banner to display an alternate UI for self-custody users.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/resources/brave_components_strings.grd [hunk 1]
```diff
@@ -549,10 +549,19 @@
 
       <message name="IDS_BRAVE_REWARDS_CONNECT_WALLET_CUSTODIAL_HEADER" desc="">Custodial accounts (requires ID verification)</message>
       <message name="IDS_BRAVE_REWARDS_CONNECT_WALLET_CUSTODIAL_TOOLTIP" desc="">Custodial accounts are accounts with trusted third-party services that hold your BAT for you.</message>
+      <message name="IDS_BRAVE_REWARDS_CONNECT_WALLET_SELF_CUSTODY_HEADER" desc="">Self-custody</message>
+      <message name="IDS_BRAVE_REWARDS_CONNECT_WALLET_SELF_CUSTODY_TOOLTIP" desc="">Self-custody gives you exclusive control over your BAT, along with complete responsibility for its security.</message>
```
Comment (mkarolin) [inline components/resources/brave_components_strings.grd:553]: Not sure the comma is needed here.

**Review by mkarolin — APPROVED** [review]
`strings`++

File: components/brave_rewards/resources/tip_panel/lib/locale_strings.ts [hunk 2]
```diff
@@ -43,7 +43,11 @@ export const localeStrings = {
   unexpectedErrorTitle: 'An unexpected error has occurred',
   unexpectedErrorText: 'Please reopen the contribution panel to try again.',
   defaultCreatorDescription: 'Welcome! Users can support this content creator by sending them contributions. It’s a way of thanking them for making great content.',
-  platformPublisherTitle: '$1 on $2'
+  platformPublisherTitle: '$1 on $2',
+  selfCustodyTitle: 'Support your favorite creators',
+  selfCustodyHeader: 'Show your love and send a token of your gratitude',
+  selfCustodyText: 'Use your web3 wallet to send a one time contribution to this creator and show them your appreciation',
+  selfCustodySendButtonLabel: 'Send with Web3 wallet'
```
Comment (emerick) [inline components/brave_rewards/resources/tip_panel/lib/locale_strings.ts:50]: Nit: One string is capitalizing Web3 and one isn't.

File: components/resources/rewards_strings.grdp [hunk 3]
```diff
@@ -526,5 +537,17 @@
...
     Welcome! Users can support this content creator by sending them contributions. It’s a way of thanking them for making great content.
   </message>
+  <message name="IDS_REWARDS_TIP_SELF_CUSTODY_TITLE" desc="">
+    Support your favorite creators
+  </message>
+  <message name="IDS_REWARDS_TIP_SELF_CUSTODY_HEADER" desc="">
+    Show your love and send a token of your gratitude
+  </message>
+  <message name="IDS_REWARDS_TIP_SELF_CUSTODY_TEXT" desc="">
+    Use your web3 wallet to send a one time contribution to this creator and show them your appreciation
```
Comment (emerick) [inline components/resources/rewards_strings.grdp:547]: Nit: See other comment on Web3 capitalization.

**Review by emerick — APPROVED** [review]

File: components/brave_rewards/resources/tip_panel/lib/locale_strings.ts (hunk 2 above)
Comment (Miyayes) [reply components/brave_rewards/resources/tip_panel/lib/locale_strings.ts:50]: We should capitalize Web3 in all cases, no matter its placement in a sentence.

File: components/resources/rewards_strings.grdp (hunk 3 above)
Comment (Miyayes) [reply components/resources/rewards_strings.grdp:547]: We should capitalize Web3 in all cases, no matter its placement in a sentence.

File: components/resources/brave_components_strings.grd (hunk 1 above)
Comment (Miyayes) [reply components/resources/brave_components_strings.grd:553]: I think we can leave comma for now. String with comma was looked at specifically by copywriters too :)

**Pull request merged**